### DISCUSSION TITLE
MissingCtorCheck for lombok @NoArgsConstructor and @AllArgsConstructor

### DISCUSSION THREAD

**Discussion by ghost** [checkstyle/checkstyle#13520] [Q&A] [answered]
When am giving @NoArgsConstructor to a service class this shows the 'Missing Ctor ' rule . found that lombok genereated the constructor for the class by decoding the class files. here is the sample code
```bash
@NoArgsConstructor
@Service
public class ReceiveMessageJsonHandler

{
	private static final String SAND = "sand";
 }
```
I have gradle(6.9) ,open jdk 13  ,springboot(2.7.0) project with following dependecy

```bash
compileOnly 'org.projectlombok:lombok:1.18.22'
annotationProcessor 'org.projectlombok:lombok:1.18.22'
testCompileOnly 'org.projectlombok:lombok:1.18.22'
testAnnotationProcessor 'org.projectlombok:lombok:1.18.22'
```

**I can't suppress this rule.expecting other solutions for this because there is no violation of the rule actually the constructor is generated the issue due to the avoiding of @generated code in the class file.**

The rule details given below
 Checks that classes (except abstract ones) define a constructor and don't rely on the default one. 
its from checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MissingCtorCheck

**Comment by rnveach** [comment, marked as answer]
Lombok is part of https://checkstyle.org/writingchecks.html#Limitations and cannot be supported by Checkstyle.

Instead of suppressing the entire rule, I recommend suppressing either this file for the check or using xpath to find a way to suppress lombok classes.
https://checkstyle.org/filters/suppressionfilter.html
https://checkstyle.org/filters/suppressionxpathfilter.html

**Reply by romani** [reply]
Example how it was suppressed for similar case https://github.com/checkstyle/checkstyle/issues/11340#issuecomment-1049455493 (different Check)

**Reply by ghost** [reply]
I have 100 files that violating this rule.Is there any shortcut for suppressing all of them easily.

**Reply by nrmancuso** [reply]
> Is there any shortcut for suppressing all of them easily.

@mariyacherian Please see https://github.com/checkstyle/checkstyle/issues/11340#issuecomment-1049455493 for the XPath suppression module that you can tweak for your needs, I can’t think of any “shorter” way of suppressing these violations than to add this one module to your config.

**Reply by romani** [reply]
@mariyacherian, please try to use chart got to get xpath query for suppression
